starky/src/proof.rs

```diff
@@ -186,31 +186,6 @@ where
     pub proof: StarkProof<F, C, D>,
 }
 
-/// A combination of STARK proofs for independent statements operating on possibly shared variables,
-/// along with Cross-Table Lookup (CTL) challenges to assert consistency of common variables across tables.
-#[derive(Debug, Clone)]
-pub struct MultiProof<
-    F: RichField + Extendable<D>,
-    C: GenericConfig<D, F = F>,
-    const D: usize,
-    const N: usize,
-> {
-    /// Proofs for all the different STARK modules.
-    pub stark_proofs: [StarkProofWithMetadata<F, C, D>; N],
-    /// Cross-table lookup challenges.
-    pub ctl_challenges: GrandProductChallengeSet<F>,
-}
-
-impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize, const N: usize>
-    MultiProof<F, C, D, N>
-{
-    /// Returns the degree (i.e. the trace length) of each STARK proof,
-    /// from their common [`StarkConfig`].
-    pub fn recover_degree_bits(&self, config: &StarkConfig) -> [usize; N] {
-        core::array::from_fn(|i| self.stark_proofs[i].proof.recover_degree_bits(config))
-    }
-}
-
```

Comment:
Are we sure we want to remove this from the API? It's a relatively convenient struct to group the whole system together. I'd rather suggest expanding it by implementing some additional helpers if needed (like extracting the aux_polys needed for the new signature above).

Reply:
I removed them simply because they are unused in the Plonky2 library. In our zkVM implementation, I changed:
`pub stark_proofs: [StarkProofWithMetadata<F, C, D>; N],`
to
`pub stark_proofs: [Option<StarkProofWithMetadata<F, C, D>>; N],`
However, it wouldn’t make sense to require every project that depends on Plonky2 to make the same changes.